### ISSUE TITLE
Error while running terraform destroy when using aws and kubernetes, helm providers

### ISSUE THREAD

**Issue by vinodhbasavani** [hashicorp/terraform#23838]
I have written the terraform modules to create all networking related resources, eks masters, eks worker nodes, kubernetes resources, helm resources.

Kubernetes and helm resources are being created only after the networking related resources are created with the dependency specified among them.

but when i run the destroy command, some of the networking resources are destroyed first so terraform unable to destroy helm or kubernetes resources since the connection to api server is failing but eks masters are still there.

As a workaround, I am removing the kubernetes and helm module state from the terraform state file and then running the terraform destroy command which is absolutely fine since upon cluster termination, all the k8 resources will be deleted.

terraform version: 
Terraform v0.12.17

**Comment by hashibot** [comment]
Hello! :wave: 

This issue doesn't seem to be following one of the standard issue templates available from [the "New Issue" page](https://github.com/hashicorp/terraform/issues/new/choose). The information requested in those templates is important for my human friends to fully understand what's going on.

Please feel free to open a new issue using either the Bug Report or Feature Request template, as appropriate, and then the Terraform maintainers will review it. Thanks!

Please note that we use GitHub issues for tracking bugs and enhancements rather than for questions. While we may be able to help with certain simple problems here it's better to use [the community forum](https://discuss.hashicorp.com/c/terraform-core), where there are more people ready to help. The GitHub issues here are generally monitored only by our few core maintainers.
